lib/ruby_lsp/requests/support/dependency_detector.rb

```diff
@@ -49,5 +49,7 @@ def direct_dependency?(gem_pattern)
         false
       end
     end
+
+    HAS_TYPECHECKER = T.let(typechecker?, T::Boolean)
```

Comment:
(nit) I'd usually expect public constants to be defined before methods.

Reply:
You could also replace the constant by memoization in `typechecker?`.

Reply:
Yeah, that would be a nice improvement.